parser/src/cfg/sexpr.rs

```diff
@@ -321,42 +442,50 @@ fn parse_with(
                         // if the stack is ever empty, return an error.
                     } = stack.pop().expect("placeholder unpopped");
                     if stack.is_empty() {
-                        return Err(Spanned::new(
-                            "Unexpected closing parenthesis".to_string(),
-                            span,
-                        ));
+                        return Err(ParseError::new(span, "Unexpected closing parenthesis"));
                     }
-                    let expr = List(Spanned::new(exprs, stack_span.cover(span.clone())));
+                    let expr = SExpr::List(Spanned::new(exprs, stack_span.cover(&span.clone())));
                     stack.last_mut().expect("not empty").t.push(expr);
                 }
-                StringTok => stack
-                    .last_mut()
-                    .expect("not empty")
-                    .t
-                    .push(Atom(Spanned::new(
-                        s[span.clone()].to_string(),
-                        span.clone(),
-                    ))),
+                StringTok => {
+                    stack
+                        .last_mut()
+                        .expect("not empty")
+                        .t
+                        .push(SExpr::Atom(Spanned::new(
+                            s[span.clone()].to_string(),
+                            span.clone(),
+                        )))
+                }
+                BlockComment => metadata.push(SExprMetaData::BlockComment(Spanned::new(
+                    s[span.clone()].to_string(),
+                    span.clone(),
+                ))),
+                LineComment => metadata.push(SExprMetaData::LineComment(Spanned::new(
+                    s[span.clone()].to_string(),
+                    span.clone(),
+                ))),
+                Whitespace => metadata.push(SExprMetaData::Whitespace(Spanned::new(
+                    s[span.clone()].to_string(),
+                    span.clone(),
```

Comment:
Looking at the definition of `ops::Index`, it looks that these clone calls (including the original unmodified one) could be removed.

Reply:
done